### ISSUE TITLE
Extending Dreambooth with Kane Wallmann's implementation

### ISSUE THREAD

**Issue by Hugo-Matias** [ShivamShrirao/diffusers#54]
[This is the relevant repo](https://github.com/kanewallmann/Dreambooth-Stable-Diffusion) I'm considering for this feature request.

It's a fork from Joe Penna's implementation, so I believe it uses checkpoint files instead of diffusers, however the implementation seems to be quite interesting and probably methodology independent.

The abstract is _simply_ replacing instance prompt and class prompt with filenames from the training dataset, both for the trained subject and regularization images. The main benefit of this seems to be that we are able to train more than one subject/class to the same model, essentially more of a fine-tuning approach. [Here's an example](https://huggingface.co/panopstor/ff7r-stable-diffusion) model using this method.

Can this be adapted to your implementation? Since this will require more steps than a regular Dreambooth session to reach satisfactory results and considering Colab's usage quotas, I think that a more streamlined training interruption and continuation would also be important to consider.

**Comment by ShivamShrirao** [comment]
Hey, I have had this in mind for some, just haven't got enough time yet to complete it.

**Comment by 1blackbar** [comment]
THat would be really great to have !

**Comment by Hugo-Matias** [comment]
Thanks Shivam, that's great news! I don't have the required expertise to be much help other than spreading the word but it's nice to know you've already considered it.

Someone posted a [similar implementation](https://www.reddit.com/r/StableDiffusion/comments/y12lzj/finetune_the_entire_stable_diffusion_model_with/) on reddit a few days ago, might be useful to have a look as well.

Since most people are training models on several different subjects/styles, having a way to train everything in a single model is desirable as managing multiple 2GB becomes wasteful and unpractical. This feels like the logical next step for Dreambooth.

**Comment by ShivamShrirao** [comment]
Added in 351f3b6206f0453706346fd34a337cdf6ac6ef07

**Comment by Hugo-Matias** [comment]
What a legend! That was lightning fast, really eager to try it out. Thanks once again.

**Comment by Hugo-Matias** [comment]
Sorry for nagging, a couple of questions popped into mind when I was training with the new method.

Is the number of class images parameter set per concept or is it the sum of all? For instance if I have a concept with a "person" class and another with a "style" class. Setting num_class_images to 500 will it pick 250 or 500 from each path?

Can we continue training if instead of pointing to huggingface we use a diffusers model saved from a previous session? Is that feasible or will it mess the model regardless of the parameters you use?

**Comment by ShivamShrirao** [comment]
@Hugo-Matias per concept.
500 from each.

You can continue training.
Depends how much it was trained and on what.

**Comment by Hugo-Matias** [comment]
Awesome, thanks for the fast reply.

Sorry for taking too much your time, forgot to ask this observation from the latest training. 

Are the concepts trained in sequence or the steps constantly switch between them? 
Let's say if I hypothetically have 2 concepts of 10 instance images each, so an epoch is 10 steps. What would happen if I set a maximum step count of 6? Does it do 3 steps on each or just 6 on the first? I'm asking this because I trained 2 persons with the same class, same amount of instance images and it looks like the second one is a bit worse in terms of resemblance. Perhaps I should train it alone to see if there is something wrong with my numbers or the dataset used.
